File: parlib_soptpure/parlib/netimpl/networkreader.py
from parlib.net.inetworkreader import INetworkReader

import socket

class NetworkReader(INetworkReader):
	def __init__(self, s):
		self.s = s
		self.f = s.makefile()

	def readAll(self):
		try:
			return self.f.read()
		except:
			return None

	def read(self):
		try:
			c = self.f.read(1)
			if c == '':
				return self.CODE_END
			return ord(c)
		except:
			return self.CODE_ERR

	def close(self):
		try:
			self.s.close()
			self.f.close()
			return True
		except socket.error:
			return False
			

File: experiments/proj_ftpclient/jclient/paskma/ftpclient/netimpl/NetworkReader.java
package paskma.ftpclient.netimpl;

import paskma.ftpclient.net.INetworkReader;

import java.io.*;
import java.net.Socket;

/**
 * User: paskma
 * Date: Oct 10, 2011
 * Time: 12:48:15 PM
 */
public class NetworkReader implements INetworkReader {
	private Socket socket;
	private BufferedReader reader;
	public NetworkReader(Socket socket) {
		this.socket = socket;
		try {
			this.reader = new BufferedReader(new InputStreamReader(socket.getInputStream()));
		} catch (IOException e) {
			e.printStackTrace();
			this.reader = null;
		}
	}

	/**
	 * For command connection.
	 */
	public String readLine() {
		if (reader == null)
			return null;

		try {
			return reader.readLine();
		} catch (IOException e) {
			e.printStackTrace();
			return null;
		}
	}

	/**
	 * For data connections, closes the socket.
	 */
	public byte[] readAll() {
		try {
			ByteArrayOutputStream result = new ByteArrayOutputStream();
			InputStream input = socket.getInputStream();
			int c;
			while ((c = input.read()) != -1) {
				result.write((byte)c);
			}

			return result.toByteArray();
		} catch (IOException ex) {
			ex.printStackTrace();
			return null;
		} finally {
			close();
		}
	}

	/**
	 * For multi-threaded data connection.
	 */
	public int read() {
		try {
			return reader.read();
		} catch (IOException e) {
			return CODE_ERR;
		}
	}

	public boolean close() {
		if (socket != null) {
			try {
				socket.close();
				return true;
			} catch (IOException e) {
				return false;
			}
		}
		return false;
	}
}
